test/test_codegen.py

```diff
@@ -174,37 +188,65 @@ def test_get_temperature(mechname):
         assert np.abs(t - t_pm) < tol
 
 
-@pytest.mark.parametrize("mechname", ["uiuc", "sanDiego"])
-def test_get_thermo_properties(mechname):
+@pytest.mark.parametrize("mechname,fuel", [("uiuc", "C2H4"),
+                                       ("sanDiego", "H2")])
+def test_kinetics(mechname, fuel):
     """This function tests that pyrometheus-generated code
-    computes thermodynamic properties c_p, s_R, h_RT, and k_eq
-    correctly by comparing against Cantera"""
-    # Create Cantera and pyrometheus objects
+    computes the right rates of progress for given temperature
+    and composition"""
     sol = ct.Solution(f"{mechname}.cti", "gas")
+    code = pyro.gen_thermochem_code(sol)
+    mechfile = open(f"{mechname}.py", "w")
+    print(code, file=mechfile)
+    mechfile.close()
+
     ptk = pyro.get_thermochem_class(sol)()
-    # Loop over temperatures
-    temp = np.linspace(500.0, 3000.0, 10)
-    for t in temp:
 
-        # Get properties from pyrometheus
-        cp_pm = ptk.get_species_specific_heats_R(t)
-        s_pm = ptk.get_species_entropies_R(t)
-        h_pm = ptk.get_species_enthalpies_RT(t)
-        keq_pm = 1 / np.exp(ptk.get_equilibrium_constants(t))
-        # Set state in cantera for comparison
-        sol.TP = t, ct.one_atm
-        keq_ct = sol.equilibrium_constants
-        # Compare properties
-        cp_err = np.abs(cp_pm - sol.standard_cp_R).max()
-        s_err = np.abs(s_pm - sol.standard_entropies_R).max()
-        h_err = np.abs(h_pm - sol.standard_enthalpies_RT).max()
-        keq_err = np.abs((keq_pm - keq_ct) / keq_ct)
-        assert cp_err < 1.0e-13
-        assert s_err < 1.0e-13
-        assert h_err < 1.0e-13
-        for i, r in enumerate(sol.reactions()):
-            if r.reversible:
-                assert keq_err[i] < 1.0e-13
+    # Homogeneous reactor to get test data
+    init_temperature = 1500.0
+    equiv_ratio = 1.0
+    ox_di_ratio = 0.21
+    stoich_ratio = 3.0  # 0.5
+    i_fu = sol.species_index(f"{fuel}")
+    i_ox = sol.species_index("O2")
+    i_di = sol.species_index("N2")
+    x = np.zeros(ptk.num_species)
+    x[i_fu] = (ox_di_ratio*equiv_ratio)/(stoich_ratio+ox_di_ratio*equiv_ratio)
+    x[i_ox] = stoich_ratio*x[i_fu]/equiv_ratio
+    x[i_di] = (1.0-ox_di_ratio)*x[i_ox]/ox_di_ratio
+    sol.TPX = init_temperature, ct.one_atm, x
```

Comment:
Is this used anywhere?

Reply:
Yes, it is used on the very next line to initialize the reactor initial state.

Reply:
Just to clarify (sol.TPX = ) sets an internal state for the _Cantera_ `sol` object.  Subsequent uses of the `sol` object depend on this data.